common/types/string_test.go

```diff
@@ -253,3 +253,190 @@ func TestStringSize(t *testing.T) {
 		t.Error("String size must be code points, not UTF8 bytes")
 	}
 }
+
+func TestStringFormat(t *testing.T) {
+	tests := []struct {
+		name           string
+		format         string
+		formatArgs     []ref.Val
+		locale         string
+		err            string
+		expectedOutput string
+	}{
+		{
+			name:           "no-op",
+			format:         "no substitution",
+			expectedOutput: "no substitution",
+		},
+
+		{
+			name:           "int substitution",
+			format:         "int is %s",
+			formatArgs:     []ref.Val{Int(1)},
+			expectedOutput: "int is 1",
+		},
+		{
+			name:           "double substitution",
+			format:         "double is %s",
+			formatArgs:     []ref.Val{Double(0.5)},
```

Comment:
I was not expecting this to be allowed.

Reply:
Yes, I think it'd make more sense with `printf` syntax to disallow this - I'll fix it.